### PR TITLE
Add page where users reviews are displayed

### PR DESCRIPTION
**Pull request by advayanand**
## Description

Frontend
- Added a reviews tab on the sidebar (/reviews) where the currently logged in user's reviews are displayed.
- Users can delete their reviews by clicking the delete button under the review they wish to delete
- Reviews page is not displayed if the user is not logged in

Backend
- Updated the DELETE /reviews endpoint to allow users to delete their own reviews

## Screenshots

https://user-images.githubusercontent.com/22901073/222995462-ba5ceb69-82ed-44d5-8484-a5271c60f4dd.mov

## Final Checks:
- [ ] Verify successful deployment
- [ ] Delete branch

(optional)
- [ ] Write tests
- [ ] Write documentation

### CODE REVIEW

File: site/src/component/UserReviews/UserReviews.tsx
Comment (ethanwong16) [inline site/src/component/UserReviews/UserReviews.tsx:45]: What is the verify review feature?  Do we want to include this in a future PR?

If so, could you add an issue and describe what the feature is, ideas for implementation, and link to this PR?

File: api/src/controllers/reviews.ts
Comment (ethanwong16) [inline api/src/controllers/reviews.ts:192]: Just a thought -- would it make more sense to say `review author` instead of `authorized user`?

I'll leave this up to you though, it's a minor nit 😅

File: api/src/controllers/reviews.ts
Comment (ethanwong16) [inline api/src/controllers/reviews.ts:175]: Just wondering, are we checking that the user IDs match just for extra assurance? 

 In theory (assuming our data isn't being intercepted / tampered with), it shouldn't be possible for the returned reviews to belong to a different user right?

**Review by ethanwong16 — APPROVED** [review]
LGTM!  Great work @advayanand 🥳 !!

I left a couple of comments and questions, but they're pretty minor and I'll go ahead and just give you approval.

It would also be great if you could document information about working with Mongo / the reviews for future devs, as well as any information you might know about the Mongo schema.

**Comment by ethanwong16** [comment]
Closes #193

File: api/src/controllers/reviews.ts
Comment (advayanand) [reply api/src/controllers/reviews.ts:192]: Done

File: site/src/component/SideBar/SideBar.tsx
Comment (advayanand) [reply site/src/component/SideBar/SideBar.tsx:92]: Done

File: api/src/controllers/reviews.ts
Comment (advayanand) [reply api/src/controllers/reviews.ts:175]: Yes, the check is just for assurance that the author of the review is the one who is trying to delete it. It shouldn't be possible for the review to belong to a different user unless the user sends a request to delete a review that doesn't belong to them.

File: site/src/component/UserReviews/UserReviews.tsx
Comment (advayanand) [reply site/src/component/UserReviews/UserReviews.tsx:45]: That was a remnant from the admin page code where admins can verify reviews before they are displayed publicly. I should have removed that line instead of commenting it out.

**Comment by ethanwong16** [comment]
Once #291 is merged into master and working, feel free to check real quick that this feature works and merge as well

**Comment by js0mmer** [comment]
Merged #291. You can go ahead and update the branch and test it out. Also don't forget to update any api calls you've added to start with /api.

**Pull request merged**
